app/screens/edit_server/index.test.tsx

```diff
@@ -0,0 +1,308 @@
+
+import EditServer from './index';
+
+import type ServersModel from '@typings/database/models/app/servers';
+
+// Mock dependencies
+jest.mock('@database/manager', () => ({
+    updateServerDisplayName: jest.fn().mockResolvedValue(undefined),
+}));
+
+jest.mock('@init/credentials', () => ({
+    getServerCredentials: jest.fn(),
+    setServerCredentials: jest.fn(),
+}));
+
+jest.mock('@managers/network_manager', () => ({
+    getClient: jest.fn(),
+}));
+
+jest.mock('@managers/websocket_manager', () => ({
+    createClient: jest.fn(),
+}));
+
+jest.mock('@screens/navigation', () => ({
+    dismissModal: jest.fn(),
+}));
+
+jest.mock('@utils/log', () => ({
+    logWarning: jest.fn(),
+}));
+
+jest.mock('@queries/app/servers', () => ({
+    getServerByDisplayName: jest.fn(),
+}));
+
+// Create mock server
+const mockServer: ServersModel = {
+    url: 'https://example.com',
+    displayName: 'Test Server',
+    lastActiveAt: Date.now(),
+} as ServersModel;
+
+const defaultProps = {
+    componentId: 'EditServer' as any,
+    server: mockServer,
+    theme: getDefaultThemeByAppearance(),
+};
+
+describe('EditServer Screen', () => {
+    let mockGetServerCredentials: jest.Mock;
+    let mockSetServerCredentials: jest.Mock;
+    let mockNetworkManagerGetClient: jest.Mock;
+    let mockWebSocketManagerCreateClient: jest.Mock;
+    let mockDismissModal: jest.Mock;
+
+    beforeEach(() => {
+        jest.clearAllMocks();
+
+        mockGetServerCredentials = require('@init/credentials').getServerCredentials;
+        mockSetServerCredentials = require('@init/credentials').setServerCredentials;
+        mockNetworkManagerGetClient = require('@managers/network_manager').getClient;
+        mockWebSocketManagerCreateClient = require('@managers/websocket_manager').createClient;
+        mockDismissModal = require('@screens/navigation').dismissModal;
+    });
+
+    it('should render edit server screen', async () => {
+        mockGetServerCredentials.mockResolvedValue({
+            token: 'test-token',
+            preauthSecret: undefined,
+        });
+
+        renderWithIntlAndTheme(<EditServer {...defaultProps}/>);
+
+        await waitFor(() => {
+            expect(screen.getByTestId('edit_server.screen')).toBeVisible();
+        });
+    });
+
+    it('should initialize with dummy value when existing password exists', async () => {
+        mockGetServerCredentials.mockResolvedValue({
+            token: 'test-token',
+            preauthSecret: 'existing-password',
+        });
+
+        renderWithIntlAndTheme(<EditServer {...defaultProps}/>);
+
+        await waitFor(() => {
+            expect(mockGetServerCredentials).toHaveBeenCalledWith('https://example.com');
+        });
+
+        // Wait a bit more for the async state update
+        await waitFor(() => {
+            const passwordInput = screen.queryByTestId('edit_server_form.preauth_secret.input');
+            if (passwordInput) {
```

Comment:
An if like this doesn't make sense in a test. This happens also in the next test.